test/e2e/basic_test.go

```diff
@@ -558,6 +478,105 @@ func getRoundNumber(data *TestData, podName string) (uint64, error) {
 	return 0, fmt.Errorf("did not find roundNum in OVSDB result")
 }
 
+// TestSyncRoutes checks that when some routes are removed, the Antrea Agent will synchronize the
+// state of routes and add missing routes back.
+func TestSyncRoutes(t *testing.T) {
+	skipIfNumNodesLessThan(t, 2)
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	if len(clusterInfo.podV4NetworkCIDR) != 0 {
+		t.Logf("Running IPv4 test")
+		testSyncRoutes(t, data, false)
+	}
+	if len(clusterInfo.podV6NetworkCIDR) != 0 {
+		t.Logf("Running IPv6 test")
+		testSyncRoutes(t, data, true)
+	}
+}
+
+func testSyncRoutes(t *testing.T, data *TestData, isIPv6 bool) {
+	encapMode, err := data.GetEncapMode()
+	if err != nil {
+		t.Fatalf(" failed to get encap mode, err %v", err)
+	}
+
+	if err != nil {
+		t.Fatalf("Failed to detect gateway interface name from ConfigMap: %v", err)
+	}
+
+	expectedRtNumMin, expectedRtNumMax := clusterInfo.numNodes-1, clusterInfo.numNodes-1
+	if encapMode == config.TrafficEncapModeNoEncap {
+		expectedRtNumMin, expectedRtNumMax = 0, 0
+
+	} else if encapMode == config.TrafficEncapModeHybrid {
+		expectedRtNumMin = 1
+	}
+	var routes []Route
+	if err := wait.PollImmediate(defaultInterval, defaultTimeout, func() (found bool, err error) {
+		routes, err = getGatewayRoutes(data, isIPv6)
+		if err != nil {
+			return false, err
+		}
+
+		if len(routes) < expectedRtNumMin {
+			// Not enough routes, keep trying
+			return false, nil
+		} else if len(routes) > expectedRtNumMax {
+			return false, fmt.Errorf("found too many gateway routes, expected %d but got %d", expectedRtNumMax, len(routes))
+		}
+		return true, nil
+	}); err == wait.ErrWaitTimeout {
+		t.Fatalf("Not enough gateway routes after %v", defaultTimeout)
+	} else if err != nil {
+		t.Fatalf("Error while waiting for gateway routes: %v", err)
+	} else {
+		t.Logf("Found all expected gateway routes")
+	}
+
+	var routeToDelete *Route
+	if encapMode.SupportsEncap() {
+		routeToDelete = &routes[0]
+	}
+	if routeToDelete != nil {
+		t.Logf("Deleting one actual gateway route")
+		if err := deleteGatewayRoute(routeToDelete, data, isIPv6); err != nil {
+			t.Fatalf("Error when deleting route: %v", err)
+		}
+	}
+
+	if err := wait.Poll(30*time.Second, 3*time.Minute, func() (bool, error) {
+		newRoutes, err := getGatewayRoutes(data, isIPv6)
```

Comment:
I am not a big fan of including a test that long as we are trying to reduce the runtime of the e2e test suite
Any chance we can rely on an integration test instead, like we did for iptables rules reconciliation: https://github.com/vmware-tanzu/antrea/pull/1751?